datalad/downloaders/http.py

```diff
@@ -293,6 +295,25 @@ def _post_credential(self, *args, **kwargs):
         raise NotImplementedError("not yet functioning, see https://github.com/kennethreitz/requests/issues/2934")
 
 
+@auto_repr
+class HTTPBearerTokenAuthenticator(HTTPRequestsAuthenticator):
+    """Authenticate via HTTP Authorization header
+    """
+    def __init__(self, **kwargs):
+        # so we have __init__ solely for a custom docstring
+        super(HTTPBearerTokenAuthenticator, self).__init__(**kwargs)
+
+    def _post_credential(self, credentials, post_url, session):
+        # we do not need to post anything, just inject token into the session
+        session.headers['Authorization'] = "Bearer %s" % credentials['token']
+        # TODO: actually access some location to verify that token is correct
```

Comment:
@driusan I wondered if there is a way to verify that the token is valid... is there?

Reply:
I don't think Authorization headers have any meaning other than what the web server receiving it gives to them, so I don't think a generic HTTPBearerTokenAuthenticator could validate the token other than by trying to make a (real) superfluous request and making sure it doesn't receive a 401 response.